### PR TITLE
Replace JUnit4 with JUnit5

### PR DESCRIPTION
**Pull request by MarcRubio**
I have replaced JUnit4 with JUnit5 with some new testCompile dependencies which keep it backwards compatible.

In order to quickly compare the difference I have duplicated RadixECKeyPairsTest naming it RadixECKeyPairsTestJUnit5

To see what annotations are available and how they compare to JUnit4 please see https://junit.org/junit5/docs/current/user-guide/#writing-tests-annotations

JUnit5 offers a bunch more of powerful features which can significantly reduce then number of code lines and keep things much neater with its annotations. 

A feature to consider is the possibility of using the Nested annotation which is basically using inner classes where Test methods use a common variable. This allows to shorten the @DisplayName even further since the "Given" part would be common and a @DisplayName can be placed on top of the inner class.

Also, in case this is missed, when tests are run the @DisplayName is shown instead of the method signatures.

### CODE REVIEW

File: radixdlt-java/src/test/java/com/radixdlt/client/core/crypto/RadixECKeyPairsTestJUnit5.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    @BeforeEach
+    public void beforeSuite() {
+        // Ensure the BouncyCastle providers are loaded into memory
+        // (because BouncyCastle SHA-256 is used in "seed" tests).
+        ECKeyPairGenerator.install();
+    }
+
+    @Test
+    @DisplayName("Given privateKey1 and privateKey2 When we compare them Then they should be different")
```
Comment (flotothemoon) [inline radixdlt-java/src/test/java/com/radixdlt/client/core/crypto/RadixECKeyPairsTestJUnit5.java]: Can we break these up over multiple lines?

Comment (MarcRubio) [reply radixdlt-java/src/test/java/com/radixdlt/client/core/crypto/RadixECKeyPairsTestJUnit5.java]: I don't believe so but come one... you should be happy, look at all those spaces you can add!

Comment (MarcRubio) [reply radixdlt-java/src/test/java/com/radixdlt/client/core/crypto/RadixECKeyPairsTestJUnit5.java]: Just a note though, to break it up, an inner class using @Nested annotation could be used and you can place the @DisplayName("Given privateKey1 and privateKey2") there on top of the inner class and keeping @DisplayName("When we compare them Then they should be different") on the method inside the inner class which will make it shorter and look nicer.

This is used when there are multiple methods using the same "Given". In this case though, there is only one method so not really necessary but would make it look neater for sure.

**Comment by mpsandiford** [comment]
Which issue is this part of?  I don't remember this being an agreed task for the sprint.

File: radixdlt-java/src/test/java/com/radixdlt/client/core/crypto/RadixECKeyPairsTestJUnit5.java (hunk 1 above)
Comment (mpsandiford) [reply radixdlt-java/src/test/java/com/radixdlt/client/core/crypto/RadixECKeyPairsTestJUnit5.java]: It's possible to use string concatenation in annotations in Java, so could be something like:
```
@DisplayName(
    "Given privateKey1 and privateKey2 " +
    "When we compare them " +
    "Then they should be different")
```

**Comment by laszlourszuly** [comment]
@MarcRubio is this something that should be merged? If not, we should consider closing it, otherwise a rebase and merge perhaps?

**Comment by MarcRubio** [comment]
I agree about doing something. This change was a non breaking and the natural upgrade to JUnit4 while keeping things compatible across the board. But since cucumber framework has been adopted for testing I think we can close it as this can be easily added in the future.

**Comment by laszlourszuly** [comment]
Closing as of ack in previous comment.

**Pull request closed without merging**